Create a map between Orca and LibXC names

felipeZ opened the pull request:
## Changes
Fix #461 by adding the ``share/packages/orca_functional_names.xml`` containing the naming correspondence between orca and libxc functionals.

JensWehner commented on share/packages/orca_functional_names.xml:
```diff
@@ -0,0 +1,7 @@
+  <XC_HYB_GGA_XC_B1LYP number="416">B1LYP</XC_HYB_GGA_XC_B1LYP>
+  <XC_HYB_GGA_XC_B3LYP number="402">B3LYP</XC_HYB_GGA_XC_B3LYP>
+  <XC_HYB_GGA_XC_PBEH number="406">PBE0</XC_HYB_GGA_XC_PBEH>
+  <XC_GGA_C_PBE number="130">PBE</XC_GGA_C_PBE>
```
this line is problemati because the PBE functional is XC_GGA_C_PBE + XC_GGA_X_PBE,

C and X indicate the correlation and exchange part separately.

felipeZ replied:
mmmm, so when you are using PBE do you call functional as:
 ```
XC_GGA_C_PBE+XC_GGA_X_PBE
```
Or how do you call it?

JensWehner replied:
XC_GGA_C_PBE XC_GGA_X_PBE

JensWehner replied:
the order does not matter

JensWehner commented on src/libxtp/qmpackages/orca.cc:
```diff
@@ -825,12 +825,38 @@ std::string Orca::WriteMethod() const {
+std::string Orca::GetOrcaFunctionalName() const {
+
+  char* votca_share = getenv("VOTCASHARE");
+  if (votca_share == nullptr) {
+    return _settings.get("functional");
```
I do not understand the logic here.

felipeZ replied:
The `VOTCASHARE` environmental variable is not defined for testing, so we need to take into account that  corner case

JensWehner replied:
yeah environment variables and testing are a shit combination. Maybe we have to think about a general solution for that. Any ideas?

JensWehner commented on src/libxtp/qmpackages/orca.cc:
```diff
@@ -825,12 +825,38 @@ std::string Orca::WriteMethod() const {
+    if (functional_names.exists(input_name)) {
+      return functional_names.get(input_name).as<std::string>();
+    } else {
+      return input_name;
```
should we not throw an error here?

felipeZ replied:
I am not sure if that is the best solution. We will throw an error based on the assumption that *Orca* doesn't know the functional that the user has passed based on an small subset of functional that we have. But I suppose that for most of the users an error in this point is more meaningful that in the Orca runtime.
I am going to change it.

felipeZ commented on src/libxtp/qmpackages/orca.cc:
```diff
@@ -825,12 +825,51 @@ std::string Orca::WriteMethod() const {
+        all_functionals.get("functionals");
+
+    std::string input_name = _settings.get("functional");
+    // Some functionals have a composed named separated by a space
```
@JensWehner  do you agree with this?

JensWehner replied:
what happens if votcashare is not set?

felipeZ replied:
Assuming that the user has command line tools in her/his path and the calculation doesn't fail because ti cannot find `xtp_tools`, `xtp_run`, etc.then that function will generate a call to Orca that will subsequently fails because it doesn't know the functional

felipeZ commented:
@JensWehner  @baumeier  Could you please review this PR?

JensWehner commented:
Should we maybe introduce a new folder which stores, qmdata default?

felipeZ commented:
> Should we maybe introduce a new folder which stores, qmdata default?

@JensWehner  I have done so at 879dc1a. I also added some README files and deleted the used `man` folder

felipeZ commented:
@JensWehner Testing the LibXC/Orca functional name conversion is difficult because of the `VOTCASHARE` environmental variable and the fact that all the new methods are private.
